src/header/common/cache_control.rs

```diff
@@ -108,8 +129,8 @@ pub enum CacheDirective {
     /// "s-maxage=delta"
     SMaxAge(u32),
 
-    /// Extension directives. Optionally include an argument.
-    Extension(String, Option<String>)
+    /// Extension directives.
+    Extension(CacheDirectiveExtension),
```

Comment:
The original goal of the `Extension` variant was to allow for new arguments even if the version of hyper doesn't support them (hence being just `String`s). Would it make sense for these new variants to just be on `CacheDirective` itself?